packages/app/pages/[modelName]/[documentTitle]/index.tsx

```diff
@@ -79,6 +87,53 @@ const EditDocumentPage = ({
         refreshDataInPlace(router)
     }, [formData.state, refreshDataInPlace])
 
+    useEffect(() => {
+        const privileges = user.privilegesForModelAndWorkflowNode(
+            modelName,
+            model.workflow.currentNode
+        )
+        // @ts-expect-error
+        const { userActivation } = globalThis.navigator
+
+        let intervalId: null | ReturnType<typeof globalThis.setInterval>
+        let consecutiveErrorCount = 0
+
+        if (userActivation) {
```

Comment:
Only run this if we can user the browser's heuristics to determine if a user has been active.